homeassistant/components/sunsynk/config_flow.py

```diff
@@ -0,0 +1,159 @@
+        if self.client is None:
+            return []
+        return await self.client.get_inverters()
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> SunsynkHub:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    if CONF_USERNAME not in data or CONF_PASSWORD not in data:
+        raise InvalidAuth
+
+    hub = SunsynkHub()
+
+    if not await hub.authenticate(data[CONF_USERNAME], data[CONF_PASSWORD]):
+        raise InvalidAuth
+
+    return hub
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Sunsynk."""
+
+    VERSION = 1
+
+    username: str | None = None
+    password: str | None = None
+    hub: SunsynkHub | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                self.hub = await validate_input(self.hass, user_input)
+                self.username = user_input[CONF_USERNAME]
+                self.password = user_input[CONF_PASSWORD]
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return await self.async_step_inverter()
+
+        return self.async_show_form(
+            step_id=STEP_USER, data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_inverter(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the inverter selection step."""
+        if user_input is None and self.hub is not None:
+            return self.async_show_form(
+                step_id=STEP_INVERTER, data_schema=await self._inverter_scheme()
+            )
+
+        assert user_input is not None
+
+        if self._inverter_already_exists(user_input[DATA_INVERTER_SN]):
+            raise AbortFlow("already_configured")
+
+        await self.async_set_unique_id(user_input[DATA_INVERTER_SN])
+        return self.async_create_entry(
+            title=f"Inverter {user_input[DATA_INVERTER_SN]}",
+            data={
+                CONF_USERNAME: self.username,
+                CONF_PASSWORD: self.password,
+                DATA_INVERTER_SN: user_input[DATA_INVERTER_SN],
+            },
+        )
+
+    def _inverter_already_exists(self, inverter_sn: str) -> bool:
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.unique_id == inverter_sn:
+                return True
+        return False
+
+    async def _inverter_scheme(self):
```

Comment:
What are we choosing here?

Reply:
We are selecting the inverter device that the user wants to connect to

Reply:
I don't own any solar panels, but what is an inverter?

Reply:
An inverter is the device that takes the DC electricity generated from solar and turns it into AC that your house can run off.

It also manages / monitors export of electricity to the grid and will provide information on the amount of energy draw coming from grid, solar and batteries (if you have battery storage as part of your solar install).

Reply:
Right, is there a reason someone wants to not add a certain inverter? What do you think the majority of the people want to do with the integration? Add all inverters?

Reply:
Most people will likely only have one inverter. 

However, you could have multiple inverters at one property for a complex setup.

If someone had multiple properties with solar they may not want to add all inverters.

Reply:
We should just add all and then the user can disable the ones they don't want/need.
Adding this selection here just adds unnecessary complexity.

Reply:
Changd to add all